operators/src/source/gdal_source.rs

```diff
@@ -820,771 +691,281 @@ mod tests {
-            [-2, -3].into(),
-            tile_size_in_pixels.into(),
-            tile_geo_transform,
-        );
-        let time_interval = TimeInterval::new_unchecked(0, 1000);
+        assert_eq!(x.data.len(), 64);
 
-        let x = gdal_source
-            .load_tile_data(time_interval, tile_information)
-            .unwrap();
-
-        assert_eq!(x.tile_information(), tile_information);
-        assert_eq!(x.time, time_interval);
+        // TODO: compare actual data
+        // assert_eq!(x.data, &[] as &[u8]);
     }
 
     #[test]
-    fn test_iter_and_load_tile_data() {
-        let global_size_in_pixels = [1800, 3600];
-        let tile_size_in_pixels = [600, 600];
-        let dataset_upper_right_coord = (-180.0, 90.0).into();
-        let dataset_x_pixel_size = 0.1;
-        let dataset_y_pixel_size = -0.1;
-        let dataset_geo_transform = GeoTransform::new(
-            dataset_upper_right_coord,
-            dataset_x_pixel_size,
-            dataset_y_pixel_size,
-        );
-
-        let bounding_box = BoundingBox2D::new((-180., -90.).into(), (180., 90.).into()).unwrap();
-
-        let origin_split_tileing_strategy = TilingStrategy {
-            tile_size_in_pixels: tile_size_in_pixels.into(),
-            geo_transform: dataset_geo_transform,
-        };
-
-        let time_interval_provider = TimeIntervalInformation {
-            start_time: TimeInstance::from_millis(0),
-            time_step: TimeStep {
-                granularity: TimeGranularity::Seconds,
-                step: 1,
-            },
-        };
-
-        let gdal_params = GdalSourceParameters {
-            dataset_id: "modis_ndvi".to_owned(),
-            channel: None,
-        };
-
-        let dataset_information = JsonDatasetInformation {
-            time: time_interval_provider,
-            tile: None,
-            file_name_with_time_placeholder: "MOD13A2_M_NDVI_2014-01-01.TIFF".into(),
-            time_format: "".into(),
-            base_path: "../modis_ndvi".into(),
-            data_type: RasterDataType::U8,
-            geo_transform: dataset_geo_transform,
-            grid_shape: global_size_in_pixels.into(),
-            spatial_ref: SpatialReference::epsg_4326().into(),
-            measurement: Measurement::Unitless,
-        };
-
-        let dataset_information_provider = JsonDatasetInformationProvider {
-            dataset_information,
-            raster_data_root: "../operators/test-data/raster".into(),
-        };
-
-        let gdal_source = GdalSourceProcessor {
-            dataset_information: dataset_information_provider,
-            gdal_params,
-            tiling_specification: TilingSpecification {
-                origin_coordinate: Coordinate2D::new(0., 0.),
-                tile_size_in_pixels: GridShape2D::from(tile_size_in_pixels),
-            },
-            phantom_data: PhantomData,
-        };
+    fn test_load_tile_data_overlaps_dataset_bounds() {
```

Comment:
is this a test or a helper? please move helpers to the top

Reply:
b321150062eb7c995d424d0eecbbb09e626a1f49